Client/Frontend/Browser/Helpers/PlaylistFolderSharingHelper.swift

```diff
@@ -0,0 +1,69 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import WebKit
+import BraveShared
+import Shared
+import Data
+
+private let log = Logger.browserLogger
+
+protocol PlaylistFolderSharingHelperDelegate: AnyObject {
+  func openPlaylistSharingFolder(with id: String)
+}
+
+class PlaylistFolderSharingHelper: NSObject, TabContentScript {
+  fileprivate weak var tab: Tab?
+  public weak var delegate: PlaylistFolderSharingHelperDelegate?
+
+  init(tab: Tab) {
+    self.tab = tab
+    super.init()
+  }
+
+  static func name() -> String {
+    return "PlaylistFolderSharingHelper"
+  }
+
+  func scriptMessageHandlerName() -> String? {
+    return "playlistFolderSharingHelper_\(UserScriptManager.messageHandlerTokenString)"
+  }
+
+  func userContentController(_ userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage, replyHandler: (Any?, String?) -> Void) {
+    defer { replyHandler(nil, nil) }
+    
+    if let sharingInfo = PlaylistFolderSharingInfo.from(message: message) {
+      // This shared playlist folder already exists
+      if let sharedFolder = PlaylistFolder.getSharedFolder(folderId: sharingInfo.playlistId) {
+        message.webView?.evaluateSafeJavaScript(functionName: "window.brave.playlist.errorHandler",
```

Comment:
We decided this will be handled on the device side, remove the js and implement it on the device side

Reply:
Fixed